### ISSUE TITLE
Use dropdown in custom UIView

### ISSUE THREAD

**Issue by leizard** [AssistoLab/DropDown#48]
Hi, thank you for an awesome package
I want to implement Dropdown in my custom view. Something like this:

```
import UIKit
import DropDown

class FiltersView: UIView {
    var view: UIView!

    @IBOutlet weak var seeViewDropDown: UIView!
    @IBOutlet weak var sortViewDropDown: UIView!
    @IBOutlet weak var filterViewDropDown: UIView!

    override init(frame:CGRect) {

        super.init(frame: frame)
        setup()
    }

    required init?(coder aDecoder: NSCoder) {

        super.init(coder: aDecoder)
        setup()
    }

    func someAction(sender:UITapGestureRecognizer){

        let dropDown = DropDown()

        // The view to which the drop down will appear on
        dropDown.anchorView = seeViewDropDown // UIView or UIBarButtonItem

        // The list of items to display. Can be changed dynamically
        dropDown.dataSource = ["Car", "Motorcycle", "Truck"]

        dropDown.show()
    }

    func setup() {
        view = loadViewFromNib()
        view.frame = bounds
        addSubview(view)

        let gestureSwift2AndHigher = UITapGestureRecognizer(target: self, action:  #selector (self.someAction(sender:)))
        seeViewDropDown.addGestureRecognizer(gestureSwift2AndHigher)
    }

    func loadViewFromNib() -> UIView {
        let bundle = Bundle(for: type(of: self))
        let nib = UINib(nibName: "FiltersView", bundle: bundle)
        view = nib.instantiate(withOwner: self, options: nil)[0] as! UIView

        return view
    }

}
```

The target is that clicking on `seeViewDropDown` view to show dropdown. But when I click `seeViewDropDown`. This error happen:

<img width="1166" alt="screen shot 2016-09-29 at 18 15 52" src="https://cloud.githubusercontent.com/assets/5291692/18951809/3fbd1a40-8671-11e6-8fbe-d092d4d6e15a.png">

I think I do something wrong but don't know where. Could you help me ?
Thank you.

**Comment by leizard** [comment]
It test Dropdown in some of mine normal ViewController but it still crashes. Seem like it doesn't related to custom view. In a IBAction of some button, I just create an instance of Dropdown and show it like this:
```
        let dropDown = DropDown()
        
        // The view to which the drop down will appear on
        dropDown.anchorView = myButton // UIView or UIBarButtonItem
        
        // The list of items to display. Can be changed dynamically
        dropDown.dataSource = ["Car", "Motorcycle", "Truck"]
        
        dropDown.show()
```

Then I click the button and it crashes

**Comment by kevin-hirsch** [comment]
Hello @leizard, thanks for using this library! :)

The solution of this problem is referenced [here](https://github.com/AssistoLab/DropDown/issues/45#issuecomment-251679881).

In short, the `dropDown` var needs to be declared as a property of your class. Otherwise, it is not retained by anything and is released at the end of the scope. In you example, `dropDown` is released at the end of the `someAction(sender:)` function.

**Comment by leizard** [comment]
@kevin-hirsch, it did work when I declared `Dropdown` as a property of my ViewController.
But now I have a different problem:
- I create a custom view to reuse in some of my main ViewController. 
- This custom view contains some buttons which are the anchorView of `Dropdown`
- What I want is to set up everything about the dropdowns in my custom view and just bring the custom view with the dropdowns to any main ViewController to use, which I can't because it will crash.
-  When the custom view instance is first created, all its buttons are nil and I think this is the cause of the crash.
- Now what I have to do to make it work is to create all the dropdown instances and set them up (datasource, anchorView, selection action,...) in every main View controller that I use my custom view in. That works but makes the code duplicate and ugly as hell.
Would appreciate if you could give my some idea  :)

**Comment by 0wnrepo** [comment]
Fixed it!

**Comment by kevin-hirsch** [comment]
Hello!
I've pushed a new commit that should fix this problem (80a125b on master)
